Description: fix AttributeError for self.focus property
Forwarded: yes wicd-devel@lists.sourceforge.net
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=709120
Author: Isis Agora Lovecruft <isis@patternsinthevoid.net>
Reviewed-By: Axel Beckert <abe@debian.org>
Last-Update: 2013-06-07

Adaptions necessary for the python-urwid 1.1 API.

 * In curses_misc.py, the class ComboBox(urwid.WidgetWrap) attempts to set set
   the attribute 'self.focus' in the method ComboBox.__init__(), though in
   urwid.WidgetWrap 'focus' in a property. Because it the 'focus' property is
   inherited, the @property methods for it must either:

     1) If the desire is to reuse the parent class, urwid.WidgetWrap, methods,
        do:

        @property
        def focus(self): return super(ComboBox, self)._get_focus()
        @property.setter
        def focus(self, index): return super(ComboBox, self)._set_focus(index)
        @property.deleter
        def focus(self): return super(ComboBox, self)._del_focus()

     2) If the desire is to create new property descriptors, do:

        def __init__(self, focus, [...]):
            [...]
            self._focus = focus

        @property
        def focus(self): return self._focus
        @property.setter
        def focus(self, index): self._focus = index
        @property.deleter
        def focus(self): del self._focus

   I went with #2.

Index: wicd-1.7.2.4/curses/curses_misc.py
===================================================================
--- wicd-1.7.2.4.orig/curses/curses_misc.py	2013-06-07 18:17:24.361894496 +0200
+++ wicd-1.7.2.4/curses/curses_misc.py	2013-06-07 18:17:24.349894443 +0200
@@ -314,9 +314,9 @@
                     #Send key to underlying widget:
                     self._w.keypress(dim, k)
 
-        #def get_size(self):
-
-    def __init__(self,label='',list=[],attrs=('body','editnfc'),focus_attr='focus',use_enter=True,focus=0,callback=None,user_args=None):
+    def __init__(self, label='', list=[], attrs=('body','editnfc'),
+                 focus_attr='focus', use_enter=True, focus=0, callback=None,
+                 user_args=None):
         """
         label     : bit of text that preceeds the combobox.  If it is "", then 
                     ignore it
@@ -349,7 +349,7 @@
         # We need this to pick our keypresses
         self.use_enter = use_enter
 
-        self.focus = focus
+        self._focus = focus
 
         self.callback = callback
         self.user_args = user_args
@@ -358,6 +358,19 @@
         self.parent = None
         self.ui = None
         self.row = None
+
+    @property
+    def focus(self):
+        return self._focus
+
+    @focus.setter
+    def focus(self, index):
+        self._focus = index
+
+    @focus.deleter
+    def focus(self):
+        del self._focus
+
     def set_list(self,list):
         self.list = list
 
